pkg/reconciler/taskrun/taskrun.go

```diff
@@ -277,15 +277,17 @@ func (c *Reconciler) finishReconcileUpdateEmitEvents(ctx context.Context, tr *v1
 
 	var err error
 	// Add status internal annotations hash only if it was verified
-	if _, notVerified := tr.Status.Annotations[spire.NotVerifiedAnnotation]; !notVerified {
-		if c.SpireClient != nil {
-			err = c.SpireClient.AppendStatusInternalAnnotation(ctx, tr)
-			if err != nil {
-				logger.Warn("Failed to sign TaskRun internal status hash", zap.Error(err))
-				events.EmitError(controller.GetEventRecorder(ctx), err, tr)
-			} else {
-				logger.Infof("Successfully signed TaskRun internal status with hash: %v",
-					tr.Status.Annotations[spire.TaskRunStatusHashAnnotation])
+	if c.SpireClient != nil {
+		if c.SpireClient.SpireVerified(tr) {
+			if cerr := c.SpireClient.VerifyStatusInternalAnnotation(ctx, tr, logger); cerr != nil {
```

Comment:
nit: combine if statements

Reply:
I think the hash compare function can be exposed and used here, not necessary to do the entire verification chain here again.